frontend/src/metabase/admin/settings/components/EmbeddingSettings/hooks.tsx

```diff
@@ -28,3 +34,21 @@ export function useMergeSetting<Key extends SettingKey>(
 
   return mergedSetting;
 }
+
+export const useEmbeddingSetting = <Key extends SettingKey>(
+  displaySetting: DisplaySetting<Key>,
+): [
+  SettingDefinition<Key>,
+  (value: SettingWithValue<Key>["value"]) => void,
+] => {
+  const mergedSetting = useMergeSetting(displaySetting);
+
+  const dispatch = useDispatch();
+
+  const handleSettingChange = (value: SettingWithValue<Key>["value"]) => {
+    dispatch(updateSetting({ key: displaySetting.key, value }));
+  };
```

Comment:
By not using `updateSetting` passed from `SettingsEditor` there won't be a toast message when settings are updated. This needs to be fixed.

Reply:
this is a really good catch. Will look into it once I resolve other comments

Reply:
I've reverted some of the things so we just use the `updateSetting` from the `SettingsEditor` for now.